### ISSUE TITLE
accessibility text for buttons / magic tap to end phone call

### ISSUE THREAD

**Issue by lilmike** [signalapp/Signal-iOS#1210]
- [x] I have searched open and closed issues for duplicates

---
### Bug description

Describe here the issue that you are experiencing.
I am blind, so use voice over on iOS to read the screen out loud. In signal, however, there are a few slight inaccessibilities plus a few improvements that coudl be made.
1. There are a few buttons that are labelled, instead of with a label like call or make a phone call, with the (I think) name of the image of the icon. For example, the phone button on the message page says 'phone-white' and the add media button says 'media-blue' or similar. Also, the group button on the contacts page shows 'group-white'.
2. In the phone call, it would be nice if voice over users coudl use the "magic tap" to end the phone call, instead of having to have the phone up to their ear to hear voice over, then finding the end button. The magic tap is a special voice over gesture which generally ends or begins something, such as music playback, a phone call, etc. It is a double tap with two fingers.
### Steps to reproduce
- using hyphens as bullet points
- list the steps
- that reproduce the bug
1. Turn on voice over, either in settings, or sometimes by tripple tapping the home button.
2. Navigate to signal, and find a chat you want to open.
3. double tap to open it, then touch the button for making a phone call, and adding media. Each will be spoken as 'phone-white' or 'media-blue'.
4. Go back, and double tap compose. You will notice, when touching the group chat button, that it is spoken as 'group-white'.
5. Go back to a message, and click the button to make a phone call. When you want to end the call, try double tapping with two fingers on the screen. It will not end the call. Instead, you will need to double tap the end button, which can be hard to find with the phone held up to your ear.

**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)
**Expected result:** Describe here what should happen after you run the steps above (i.e. what would be the correct behaviour)
### Device info

**Device**: iPhone 5s.
**iOS version**: 9.2

**Comment by cannona** [comment]
I am also a VO user, and so am also effected by this issue.  If there is anything I can do to help get this resolved, please don't hesitate to ask.

TL;DR: me too!

**Comment by paddyblifu** [comment]
I' also a vo-user too and I've found the following issue:

I'm using the newest signal iphone app on a iphone 6 (IOS 9.3.4). When I get messages from a person who is saved as a contact in my phone, voice over reads not their name, it reads the phone number before the message. This is very confusing, because in group chats with more members. When the iphone is blocked and tere's a message, voice over reads it correctly with the name before the message. For example, when I get a message in a group from a contact named "Max", voice over reads not "Max, hellow Patrick", it reads "+49160xxxxx hellow Patrick." The problem is in  the logscreen, how it is with pop-ups I don't know yet, It's with all my contact's, not with a special one.

**Comment by paddyblifu** [comment]
vo does read phone numbers insteat of contact names

**Comment by qtatspi** [comment]
I'm running latest signal on IOS 10 with an Iphone 5 c, and notice the same issues, plus one rather confusing issue, when attempting to compose a message to someone, searching by number or name, brings up nothing, just a cancel button, and nowhere can I using Voiceover as well, find an invite button, contacts is allowed under settings>signal, but searching returns nothing, and it appears I can't invite a non-signal contact to the app

**Comment by michaelkirk** [comment]
I think this:
> when attempting to compose a message to someone, searching by number or name, brings up nothing, just a cancel button, and nowhere can I using Voiceover as well, find an invite button, contacts is allowed under settings>signal, but searching returns nothing, and it appears I can't invite a non-signal contact to the app

Is covered here:
https://github.com/WhisperSystems/Signal-iOS/issues/688

**Comment by MarcoZehe** [comment]
These issues are still present in the latest Signal for iOS. All buttons that are comprised of images instead of text need the AccessibilityLabel provided, preferably by calling SetAccessibilityLabel with an NSLocalizedString on the button object. One could also do this via Interface Builder in Xcode, but the moment one changes the asset or image for the button, one has to re-do that. Doing it in code is safer since the missing label will be yelled out when compiling. ;-)

The VoiceOverMagicTap (two finger double tap) to end a call is done by implementing the VoiceOverMagicTap method on the appropriate delegate or in the main app delegate. Should end the phone call when one is in progress, and not pass through to super in that case, and in all other cases, just pass through to super, which is whichever default iOS then assigns to it.

**Comment by jschneier** [comment]
@MarcoZehe is the magic tap generally only used for hanging up phone calls or is it also used for answering them as well?

**Comment by cannona** [comment]
Apparently both, though answering has never worked consistantly for me.

**Comment by MarcoZehe** [comment]
It is also used for answering them.

Von meinem iPhone gesendet

**Comment by paddyblifu** [comment]
Hi,

today i'm noticed that the microfone-button, were you can record a voice
message isn't available with voice over or is it possible to record a
voice message with another way?

Thanks for fixing this problem.

regards,
Patrick


Am 13.11.2016 um 12:10 schrieb Marco Zehe:

**Comment by riyapenn** [comment]
@paddyblifu This is not yet a feature to send voice notes. See https://github.com/WhisperSystems/Signal-iOS/issues/230

**Comment by MarcoZehe** [comment]
Has the MagicTap ever been implemented for Signal voicer calls? The other buttons are now all labeled. Can this issue be closed?
